projects/Amlogic/patches/ffmpeg/ffmpeg-999-z-fixes.patch

```diff
@@ -0,0 +1,51 @@
+diff --git a/libavcodec/v4l2_m2m_dec.c b/libavcodec/v4l2_m2m_dec.c
+index 80a09f7a43..af65927ac3 100644
+--- a/libavcodec/v4l2_m2m_dec.c
++++ b/libavcodec/v4l2_m2m_dec.c
+@@ -125,6 +125,8 @@ static int v4l2_prepare_decoder(V4L2m2mContext *s)
+     return 0;
+ }
+ 
++static AVPacket saved_avpkt = { 0 };
++
+ static int v4l2_receive_frame(AVCodecContext *avctx, AVFrame *frame)
+ {
+     V4L2m2mContext *s = ((V4L2m2mPriv*)avctx->priv_data)->context;
+@@ -133,9 +135,14 @@ static int v4l2_receive_frame(AVCodecContext *avctx, AVFrame *frame)
+     AVPacket avpkt = {0};
+     int ret;
+ 
+-    ret = ff_decode_get_packet(avctx, &avpkt);
+-    if (ret < 0 && ret != AVERROR_EOF)
+-        return ret;
++    if (saved_avpkt.size) {
++	avpkt = saved_avpkt;
++	memset(&saved_avpkt, 0, sizeof(saved_avpkt));
++    } else {
++        ret = ff_decode_get_packet(avctx, &avpkt);
++        if (ret < 0 && ret != AVERROR_EOF)
++            return ret;
++    }
+ 
+     if (s->draining)
+         goto dequeue;
+@@ -144,6 +151,8 @@ static int v4l2_receive_frame(AVCodecContext *avctx, AVFrame *frame)
+     if (ret < 0) {
+         if (ret != AVERROR(ENOMEM))
+            return ret;
++
++        saved_avpkt = avpkt;
+         /* no input buffers available, continue dequeing */
```

Comment:
I'm not 100% sure, but believe the original author of that patch was @Elyotna when we first got the Amlogic V4L2 driver working.

Reply:
Yes, that was a quick patch I made a long time ago to fix the following issue:
- FFmpeg retrieves a packet from the bitstream
- It attempts to get an input buffer (from its own list or by dequeuing one from the driver)
- If no input buffer is found, the bitstream packet is dropped instead of scheduled for trying again later

Reply:
@Elyotna Awesome, thank you for the context! I can replicate this issue as well with `-f null`, which means decoding via v4l2 has been randomly broken in the past couple ffmpeg releases.

I cleaned up your patch to remove the global variable and pulled it into my patchset: https://github.com/tmm1/FFmpeg/commit/80ba42c5d624529a1a5d2eb030b8a33e92236fa5

cc @ldts